spring-data-r2dbc/src/main/java/org/springframework/data/r2dbc/dialect/PostgresDialect.java

```diff
@@ -58,6 +58,18 @@ public class PostgresDialect extends org.springframework.data.relational.core.di
 				"io.r2dbc.postgresql.codec.Polygon") //
 				.forEach(s -> ifClassPresent(s, simpleTypes::add));
 
+		// support the native JTS types supported by r2dbc-postgresql
+		Stream.of("org.locationtech.jts.geom.Geometry", //
```

Comment:
It is sufficient to just add `Geometry` as the other types are subclasses and therefore considered simple types.